src/components/FormValidator.js

```diff
@@ -79,7 +79,11 @@ export default class FormValidator {
     this._formElement.addEventListener("submit", (evt) => {
       evt.preventDefault();
       this._formElement.reset();
-      this._disableButton();
+      if (this._formElement.querySelector("#confirmation-button")) {
+        this._enableButton();
+      } else {
+        this._disableButton();

```

Comment:
Please, delete the whole code from line 81 till 85 from `FormValidator.js`

You need to **reset** the form only in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

--------------


You need to disable the button only in block `then`, otherwise you'll not be able to click it if there is an error